### PR TITLE
Karma to Jest: 3 data sources

### PR DESCRIPTION
**Pull request by tskarhed**
Ref #12224 
Converts data source tests for:
- PostgreSQL
- MySQL
- Cloudwatch

### CODE REVIEW

**Review by marefr — COMMENTED** [review]
Lets wait with this until we've settled how to move forward in #12493

**Comment by marefr** [comment]
#12493 is merged so now you can rebase this on master. and I'll review it

File: public/app/plugins/datasource/cloudwatch/specs/datasource.jest.ts [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
 import 'app/features/dashboard/time_srv';
+import * as dateMath from 'app/core/utils/datemath';
 
 describe('CloudWatchDatasource', function() {
-  var ctx = new helpers.ServiceTestContext();
-  var instanceSettings = {
+  let instanceSettings = {
     jsonData: { defaultRegion: 'us-east-1', access: 'proxy' },
   };
+  let templateSrv = new TemplateSrvStub();
```
Comment (marefr) [inline public/app/plugins/datasource/cloudwatch/specs/datasource.jest.ts]: I think you can use jest mock here like you've did for postgres, mysql and recently elasticsearch. Or is there something I'm missing?

**Review by marefr — CHANGES_REQUESTED** [review]
Just a minor comment. Besides that it looks good

File: public/app/plugins/datasource/cloudwatch/specs/datasource.jest.ts (hunk 1 above)
Comment (dehrax) [reply public/app/plugins/datasource/cloudwatch/specs/datasource.jest.ts]: It seems like I forgot that! Thanks!

**Review by davkal — APPROVED** [review]
All looks great now! Some notes still:

1 ) You don't need the mock function wrapper `jest.fn()` if you don't test for the wrapped method's invocation, see https://jestjs.io/docs/en/mock-functions#using-a-mock-function
A couple of lines can thus be simplified:
```js
variableExists: jest.fn(() => false),    ====>    variableExists: () => false,
```
 
2 ) The cloudwatch test suite still imports `'app/features/dashboard/time_srv';`. Is that needed?

**Pull request merged**